Content.Shared/Mindshield/FakeMindShield/SharedFakeMindShieldImplantSystem.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Content.Shared.Actions;
+using Content.Shared.Implants.Components;
+using Content.Shared.Mindshield.Components;
+
+namespace Content.Shared.Mindshield.FakeMindShield;
+
+public abstract class SharedFakeMindShieldImplantSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SubdermalImplantComponent, FakeMindShieldToggleEvent>(OnFakeMindShieldToggle);
+    }
+    /// <summary>
+    /// Raise the Action of a Implanted user toggling their implant to the FakeMindshieldComponent on their entity
+    /// </summary>
+    private void OnFakeMindShieldToggle(EntityUid uid,
+        SubdermalImplantComponent component,
+        FakeMindShieldToggleEvent ev)
+    {
+        ev.Handled = true;
+        if (component.ImplantedEntity is not { } ent)
+            return;
+
+        if (!TryComp<FakeMindShieldComponent>(ent, out var comp))
+            return;
+        _actionsSystem.SetToggled(ev.Action, !comp.IsEnabled); // Set it to what the Mindshield component WILL be after this
+        RaiseLocalEvent(ent, ev);
```

Comment:
The reason for it being an event is for the later use by Changeling to be able to have a different action that Directly interacts with the FakeMindshieldComponent rather than via the subdermalimplant and having it relay.